test/unittest/xmlparser/XMLElementParserTests.cpp

```diff
@@ -3863,3 +3875,205 @@ TEST_F(XMLParserTests, getXMLTypeLookupSettings_NegativeClauses)
     EXPECT_EQ(XMLP_ret::XML_ERROR,
             XMLParserTest::getXMLTypeLookupSettings_wrapper(titleElement, settings, ident));
 }
+
+TEST_F(XMLParserTests, get_element_text)
+{
+    using namespace eprosima::fastdds::xml::detail;
+
+    // 1. Empty content
+    {
+        tinyxml2::XMLDocument xml_doc;
+        tinyxml2::XMLElement* xml_element;
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse("<elem></elem>"));
+        xml_element = xml_doc.RootElement();
+
+        std::string result;
+        EXPECT_EQ(get_element_text(xml_element), "");
+        EXPECT_FALSE(get_element_text(xml_element, result));
+    }
+
+    // 2. Plain content
+    {
+        tinyxml2::XMLDocument xml_doc;
+        tinyxml2::XMLElement* xml_element;
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse("<elem>Content</elem>"));
+        xml_element = xml_doc.RootElement();
+
+        std::string result;
+        EXPECT_EQ(get_element_text(xml_element), "Content");
+        EXPECT_TRUE(get_element_text(xml_element, result));
+        EXPECT_EQ(result, "Content");
+    }
+}
+
+static void set_environment_variable(
+        const char* const env_var,
+        const char* const value)
+{
+#ifdef _WIN32
+    ASSERT_EQ(0, _putenv_s(env_var, value));
+#else
+    ASSERT_EQ(0, setenv(env_var, value, 1));
+#endif // _WIN32
+}
+
+static void clear_environment_variable(
+        const char* const env_var)
+{
+#ifdef _WIN32
+    ASSERT_EQ(0, _putenv_s(env_var, ""));
+#else
+    ASSERT_EQ(0, unsetenv(env_var));
+#endif // _WIN32
+}
```

Comment:
Please consider moving this to a unit tests utilities header file and use it in the XML Profile Parser tests as well, I think it will improve readability

Reply:
9507762